hail/src/test/scala/is/hail/expr/ir/BlockMatrixIRSuite.scala

```diff
@@ -158,4 +158,9 @@ class BlockMatrixIRSuite extends SparkSuite {
     assertBmEq(actualOnesDivRowOnLeft.execute(hc),  expectedOnesDivRowLeft)
     assertBmEq(actualOnesDivColOnLeft.execute(hc),  expectedOnesDivColLeft)
   }
+
+  @Test def testBlockMatrixDot() {
+    val dotTwosAndThrees = BlockMatrixDot(new BlockMatrixLiteral(twos), new BlockMatrixLiteral(threes))
+    assertBmEq(dotTwosAndThrees.execute(hc), BlockMatrix.fill(hc, 3, 3, 2 * 3 * 3))
```

Comment:
Why is this `2 * 3 * 3` and not `2 * 3`?

Reply:
It's a matrix multiplication that's not element-wise. So 2 * 3 summed over a dimension of length 3.